includes/class-wcpdf-main.php

```diff
@@ -847,35 +851,83 @@ public function set_header_logo_height( $document_type, $document = null ) {
 	}
 
 	/**
-	 * Remove attachments older than 1 week (daily, hooked into wp_scheduled_delete )
+	 * Schedule temporary files cleanup from paths older than 1 week (daily, hooked into wp_scheduled_delete )
 	 */
-	public function attachments_cleanup() {
-		if ( !function_exists("glob") || !function_exists('filemtime') ) {
-			// glob is required
-			return;
+	public function schedule_temporary_files_cleanup() {
+		$cleanup_age_days = isset( WPO_WCPDF()->settings->debug_settings['cleanup_days'] ) ? floatval( WPO_WCPDF()->settings->debug_settings['cleanup_days'] ) : 7.0;
+		$delete_timestamp = time() - ( intval ( DAY_IN_SECONDS * $cleanup_age_days ) );
+		$this->temporary_files_cleanup( $delete_timestamp );
+	}
+	
+	/**
+	 * Temporary files cleanup from paths
+	 * @param  int    $delete_timestamp timestamp of the date/time before which to clean up files
+	 * 
+	 * @return array  Output message
+	 */
+	public function temporary_files_cleanup( $delete_timestamp ) {
+		global $wp_filesystem;
+
+		$delete_before = ! empty( $delete_timestamp ) ? intval( $delete_timestamp ) : time();
+
+		if ( ! function_exists( 'WP_Filesystem' ) ) {
+			require_once( ABSPATH . 'wp-admin/includes/file.php' );
 		}
 
-		
-		if ( !isset( WPO_WCPDF()->settings->debug_settings['enable_cleanup'] ) ) {
-			return;
+		if ( ! WP_Filesystem() ) {
+			return array( 'error' => esc_html__( "WP_Filesystem couldn't be initiated!", 'woocommerce-pdf-invoices-packing-slips' ) );
 		}
 
+		$paths_to_cleanup = apply_filters( 'wpo_wcpdf_cleanup_tmp_paths', array(
+			$this->get_tmp_path( 'attachments' ),
+			$this->get_tmp_path( 'dompdf' ),
+		) );
+		$excluded_files   = apply_filters( 'wpo_wcpdf_cleanup_excluded_files', array(
+			'index.php',
+		) );
+		$folders_level    = apply_filters( 'wpo_wcpdf_cleanup_folders_level', 3 );
+		$files            = array();
+		$success          = 0;
+		$error            = 0;
+		$output           = array();
+
+		foreach ( $paths_to_cleanup as $path ) {
+			$files = array_merge( $files, list_files( $path, $folders_level ) );
+		}
 
-		$cleanup_age_days = isset(WPO_WCPDF()->settings->debug_settings['cleanup_days']) ? floatval(WPO_WCPDF()->settings->debug_settings['cleanup_days']) : 7.0;
-		$delete_timestamp = time() - ( intval ( DAY_IN_SECONDS * $cleanup_age_days ) );
+		if ( ! empty( $files ) ) {
+			foreach ( $files as $file ) {
+				$filetype = wp_check_filetype( $file ); // 'php' files return 'false' for 'ext'
```

Comment:
which means that `php` files are skipped here:
https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/80d67e00e6d16b396e9b25b7bd663e49e05e7ff0/includes/class-wcpdf-main.php#L903

I'm not sure why you call `wp_check_filetype` if you only check for `false` and then even skip when it's false. What's the reason for this?

Several other files were skipped too in my tests (`.json`, `.dat`)